airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -179,6 +179,7 @@ def _get_conn_params(self) -> Dict[str, Optional[str]]:
             "role": self.role or role,
             "authenticator": self.authenticator or authenticator,
             "session_parameters": self.session_parameters or session_parameters,
+            "application": "AIRFLOW",
```

Comment:
I was hoping we could use: `AIRFLOW_SNOWFLAKE_PARTNER` or a similar environment variable so even the partners don't need to change the provider, what do you think?

Example:

```suggestion
            "application": os.environ.get("AIRFLOW_SNOWFLAKE_PARTNER", "AIRFLOW"),
```

So it default to `AIRFLOW` and your partners just need to add that Environment variable in their image, example:

```
AIRFLOW_SNOWFLAKE_PARTNER=AWS
```

or 

```
AIRFLOW_SNOWFLAKE_PARTNER=Astronomer
```

or

```
AIRFLOW_SNOWFLAKE_PARTNER=GCP
```